pkg/proc/bininfo.go

```diff
@@ -1756,3 +1756,36 @@ func (bi *BinaryInfo) symLookup(addr uint64) (string, uint64) {
 	}
 	return "", 0
 }
+
+// ImportPathToDirectoryPath returns a list a mapping from pacakge
+// import paths (such as reported by the ImportPath field of 'go list') to
+// the package directory paths (i.e. the directory that contained the
+// package when the binary was created).
+func (bi *BinaryInfo) ImportPathToDirectoryPath() [][2]string {
+	m := make(map[string]string)
+	for _, cu := range bi.compileUnits {
+		if cu.image != bi.Images[0] || !cu.isgo || cu.lineInfo == nil {
+			//TODO(aarzilli): what's the correct thing to do for plugins?
+			continue
+		}
+
+		ip := strings.Replace(cu.name, "\\", "/", -1)
+		if _, ok := m[ip]; ok {
+			continue
+		}
+		path := cu.lineInfo.FirstFile()
+		if ext := filepath.Ext(path); ext != ".go" && ext != ".s" {
+			continue
+		}
+		dp := filepath.Dir(path)
+		m[ip] = dp
```

Comment:
Use of map here implies one to one correspondence between import path and directory path. If I understand correctly this might not be the case for bazel builds: https://github.com/bazelbuild/intellij/blob/master/golang/src/com/google/idea/blaze/golang/resolve/BlazeGoPackage.java#L78.

Reply:
That's annoying, I don't think there's a way to do this for bazel builds. We could also return a list of all files that partecipated on the build of each package, then the IDE could use the directory guess in modules mode and the file list in bazel mode.

Reply:
I think any of the following will be fine for IDE: a list of files per import path, or a list of directories per import path, or the current solution but changed to allow several pairs with the same import path but different directories.